src/Propel/Generator/Behavior/Versionable/VersionableBehaviorObjectBuilderModifier.php

```diff
@@ -246,22 +251,49 @@ public function isVersioningNecessary(\$con = null)
     }
 ";
         }
-        $plural = true;
+
         foreach ($this->behavior->getVersionableReferrers() as $fk) {
-            $fkGetter = $this->builder->getRefFKPhpNameAffix($fk, $plural);
-            $script .= "
-    // to avoid infinite loops, emulate in save
-    \$this->alreadyInSave = true;
-    foreach (\$this->get{$fkGetter}(null, \$con) as \$relatedObject) {
-        if (\$relatedObject->isVersioningNecessary(\$con)) {
-            \$this->alreadyInSave = false;
 
+            if ($fk->isLocalPrimaryKey()) {
+
+                $fkGetter = $this->builder->getRefFKPhpNameAffix($fk);
+                $script .= "
+    if (\$this->single{$fkGetter}) {
+
+        // to avoid infinite loops, emulate in save
+        \$this->alreadyInSave = true;
+
+        if (\$this->single{$fkGetter}->isVersioningNecessary(\$con)) {
+
+            \$this->alreadyInSave = false;
```

Comment:
@marcj I searched intensively why this code ... and I could not understand ... Can you help me?

I found this Issues ... 

- #77 
- propelorm/Propel#205


In the both case(1:1 or 1:n) when remove $this->alreadyInSave...  Bang! SegFault

![captura de tela de 2016-06-18 16-14-51](https://cloud.githubusercontent.com/assets/2307808/16173182/e592ca44-356f-11e6-88d7-913ebc546574.png)

So the code is in loop on lines 5111 and 2006.